dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -362,14 +363,28 @@ public void prepare() {
     }
 
     public Boolean betaFenceCheck() {
-        if (isBetaEdition == null) {
-            if (Boolean.getBoolean("com.ibm.ws.beta.edition")) {
-                isBetaEdition = true;
-            } else {
-                isBetaEdition = false;
+        if (isBetaEdition != null)
+            return isBetaEdition;
+
+        if (issuedBetaMessageAccess) {
+            return false;
```

Comment:
So, would there ever be a multi-threaded scenario where you have IsBetaEdition is true, while issuedBetaMessageAccess is true, the other thread might return false, which means the beta code won't get run?